Ne resize la carte que lorsque nécessaire

jillro opened the pull request:
Utiliser `requestAnimationFrame` devrait régler les erreurs `ResizeObserverLoop`

jillro committed:
feat: fix ResizeObserver loop errors

Signed-off-by: Maud Royer <hello@maudroyer.fr>

thom4parisot commented on src/components/Map/MapContainer.vue:
```diff
@@ -55,8 +55,14 @@ watch(() => props.style, () => {
 
-useResizeObserver(mapContainer, () => {
-  map.value.resize()
+useResizeObserver(mapContainer, (entries) => {
```
Je me demande si quand on place le `useResizeObserver` dans le `map.once('load')`, on aboutit au même résultat.

thom4parisot committed:
fix(map): déclenche le redimensionnement après le rendu

Ainsi, c'est un événement d'interface qui le déclenche plutôt qu'une observation permanente d'un effet de clic.

thom4parisot commented:
Alors au final, le `map.value.resize()` génère un changement de dimension dans la modale, donc l'événement du `useResizeObserver()` est constamment relancé (le bug persistait).

Ce que je me raconte, c'est que le changement de "mode" (compact ou fullSize) qui devrait trigger le refresh. Y'a un peu une condition d'exécution — il faut que le clic aie eu lieu, que le composant se soit agrandi, et là on peut déclencher le redimensionnement. C'est pas immédiat au changement de valeur d'une propriété.
